flink-core/src/main/java/org/apache/flink/api/common/io/FileInputFormat.java

```diff
@@ -461,8 +463,9 @@ public LocatableInputSplitAssigner getInputSplitAssigner(FileInputSplit[] splits
 		
 		// take the desired number of splits into account
 		minNumSplits = Math.max(minNumSplits, this.numSplits);
-		
-		final Path path = this.filePath;
+
+		final Path path = new Path(URLDecoder.decode(this.filePath.toString(), Charset.defaultCharset().name()));
```

Comment:
An awesome solution always welcomed.  I can literally use ```parentDir.toString()``` to avoid that issue.  Not to call ```parentDir.toURI().toString().```